### DISCUSSION TITLE
Add support for archive.ph

### DISCUSSION THREAD

**Discussion by sanello84** [Kovah/LinkAce#669] [Ideas & Feature Requests]
As title says. Wayback machine has historically removed pages and implementing an alternative would be great!

**Comment by chris-fj** [comment]
Came to request this. What would it take to implement the archival on archive.ph or archive.is?

**Reply by Kovah** [reply]
A simple API that LinkAce can use to trigger the archiving of URLs. As far as I know, archive.ph has no API for submitting sites, only for searching versions.

I would imagine something like `archive.ph/api/v1/save?url=https://wsj.com/bla....` and it returns a 200 with some job ID, similar to what the Internet Archive does.

**Reply by chris-fj** [reply]
I was reading the [FAQ](https://archive.is/faq) of archive.is, which seems to be a mirror of archive.ph and some others (on the section is Does it support any API?) and they seem to have an API. However, I am too dumb today because I read it and I don't quite understand it. Maybe can you check if you can make any sense of it? And thanks for answering

**Reply by chris-fj** [reply]
I reply myself. I found [this discussion](https://groups.google.com/g/memento-dev/c/XVAVyqyr9xQ?pli=1) ([mirror](https://archive.is/owjiT)), it seems to be promising, the gist of it is that there's an endpoint /submit that takes the URL as parameter and returns an output (ok/ko) as well as the archived URL.

**Reply by Kovah** [reply]
Yeah as I said, it has an API, but not for saving URLs, just for looking up versions of a URL (Memento API).
It seems to be the same with the Internet Archive. They also have no real API for saving URLs, but at least provide machine-readable JSON output for the saving endpoint. But it's all far from ideal. 

I wish there would be some decentralized system for archiving **web** content. The Internet Archive is nice, but they continiously make bad headlines with other archivings, because they are far too idealistic (book lending during Corona, for example). On the other side, who the hell is behind Archive.ph? There's no information about the service, no financial info, it could be shut down tomorrow and no one would know anything. You can say what you want, but I would like to be somewhat sure that an archive will be there for at least another decade, financial stable and well maintained.

**Reply by chris-fj** [reply]
I had some trouble replicating the experiments in the linked pages above because of the captcha and because I am behind a corporate network right now. Maybe playing with user agents will help here. However, from the title of the post, the semantics (endpoint called `/submit/`) and the positive reply from OP, the solution shown there seems to fulfill the purpose, which is getting the last URL if there exists one and creating one if not. About the response, I agree that there's no JSON output but one solution I would think that curl output can be massaged with the option `--write-out` to have a consistent format. Just throwing out ideas here.

Regarding your second paragraph, I agree with your feeling. I think the only safe backups are the ones you do, because things that are archived by others will depend on the stakeholders funding it. There's this initiative of self hosted archival solution (https://github.com/ArchiveBox/ArchiveBox/wiki), which seems to have a beta version of their REST API, exactly what we need (https://github.com/ArchiveBox/ArchiveBox/issues/496#issuecomment-2080174235). However, I understand few will be setting up a dedicated archive box so I understand that supporting this won't be seen as a priority.

**Reply by Kovah** [reply]
I will have a look at this once I work on integrating more archives.

Actually, there is a request to add support for Archivebox (#496) and I am following the issue on the Archivebox repo you linked. I too support hosting your own archive and as soon as Archivebox has a stable API, I will take a look at it.
